src/status_im/ens/core.cljs

```diff
@@ -0,0 +1,163 @@
+(ns status-im.ens.core
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.accounts.update.core :as accounts.update]
+            [status-im.ethereum.abi-spec :as abi-spec]
+            [status-im.ethereum.contracts :as contracts]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.ens :as ens]
+            [status-im.ethereum.resolver :as resolver]
+            [status-im.ethereum.stateofus :as stateofus]
+            [status-im.ui.screens.navigation :as navigation]
+            [status-im.utils.fx :as fx]
+            [status-im.utils.money :as money]
+            [status-im.wallet.core :as wallet])
+  (:refer-clojure :exclude [name]))
+
+(defn name [custom-domain? username]
+  (if custom-domain?
+    username
+    (stateofus/subdomain username)))
+
+(re-frame/reg-fx
+ :ens/resolve-address
+ (fn [[registry name cb]]
+   (ens/get-addr registry name cb)))
+
+(re-frame/reg-fx
+ :ens/resolve-pubkey
+ (fn [[registry name cb]]
+   (resolver/pubkey registry name cb)))
+
+(defn assoc-state-for [db username state]
+  (assoc-in db [:ens/registration :states username] state))
+
+(defn assoc-details-for [db username k v]
+  (assoc-in db [:ens/registration :details username k] v))
+
+(defn assoc-username-candidate [db username]
+  (assoc-in db [:ens/registration :username-candidate] username))
+
+(defn empty-username-candidate [db] (assoc-username-candidate db ""))
+
+(fx/defn set-state
+  {:events [:ens/set-state]}
+  [{:keys [db]} username state]
+  {:db (assoc-state-for db username state)})
+
+(defn- on-resolve [registry custom-domain? username address public-key s]
+  (cond
+    (= (ethereum/normalized-address address) (ethereum/normalized-address s))
+    (resolver/pubkey registry (name custom-domain? username)
+                     (fn [ss]
+                       (if (= ss public-key)
```

Comment:
Maybe it makes sense to move the conditional inside `dispatch`'s argument vector.

Reply:
:+1: